woodwork/type_sys/inference_functions.py

```diff
@@ -82,6 +93,23 @@ def timedelta_func(series):
     return False
 
 
+def num_common_words(wordlist: Union[Tokens, Any]) -> float:
+    if not isinstance(wordlist, Iterable):
+        return np.nan
+    num_common_words = 0
+    for x in wordlist:
```

Comment:
I don't know if a list comprehension would be faster than a loop here, but I think we could get the number of common words like this:
```
num_common_words = len([x for x in wordlist if x.lower() in COMMON_WORDS_SET])
```

Reply:
I wrote it this way for performance but I don't feel strongly one way or another. Guess the question is if we want to save 3 lines for 14% runtime

```python
from woodwork.type_sys.inference_functions import COMMON_WORDS_SET, NL_delimiters
from woodwork.demo import load_retail

df = load_retail()
series = df['description']
tokens = series.astype("string").str.split(NL_delimiters)

def loop(wordlist):
    num_common_words = 0
    for x in wordlist:
        if x.lower() in COMMON_WORDS_SET:
            num_common_words += 1
    return num_common_words

def list_comprehension(wordlist):
    num_common_words = len([x for x in wordlist if x.lower() in COMMON_WORDS_SET])
    return num_common_words

%timeit -n 100 tokens.map(loop)
%timeit -n 100 tokens.map(list_comprehension)
```
327 ms ± 2.13 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)
373 ms ± 1.36 ms per loop (mean ± std. dev. of 7 runs, 100 loops each)

Reply:
Go with the option that has the best performance. I wasn't trying to reduce lines of code with my suggestion, but I was trying to find the option with the best performance and wasn't sure which it would be. I think generally for our inference functions, performance should always trump a reduction in lines of code.

Reply:
Resolving this comment